src/GraphQL/Types/IProvideMetadata.cs

```diff
@@ -10,10 +10,15 @@ namespace GraphQL.Types
     /// </summary>
     public interface IProvideMetadata
     {
+        /// <summary>
+        /// Provides all directives applied to this provider.
+        /// </summary>
+        AppliedDirectives AppliedDirectives { get; }
```

Comment:
Directives can be applied to any element of the schema (even to directives 😄  ). Generally speaking, this is meta information. We can get directives from `Metadata` property indeed. I hesitated for a long time whether to do so. So far I've made a separate property.

Reply:
Done.